app/app.go

```diff
@@ -372,8 +379,27 @@ func NewEthermintApp(
 		appCodec, keys[feemarkettypes.StoreKey], app.GetSubspace(feemarkettypes.ModuleName),
 	)
 
-	// Create DXNS keepers
-	app.BondKeeper = bondkeeper.NewKeeper(appCodec, app.AccountKeeper, app.BankKeeper, keys[bondtypes.StoreKey], app.GetSubspace(bondtypes.ModuleName))
+	// Create Vulcanize dxns keepers
+
+	app.AuctionKeeper = auctionkeeper.NewKeeper(
+		app.AccountKeeper, app.BankKeeper, keys[auctiontypes.StoreKey],
+		appCodec, app.GetSubspace(auctiontypes.ModuleName),
+	)
+
+	app.NameServiceRecordKeeper = nameservicekeeper.NewRecordKeeper(app.AuctionKeeper, keys[nameservicetypes.StoreKey], appCodec, *cdc)
```

Comment:
Let's avoid legacy codec usage. @gsk967 do we have any other dependencies to keep legacy codec here? @ashwinphatak any thoughts?

Reply:
Agreed, let's avoid using the legacy codec. @gsk967 is there some issue removing it?

Reply:
Yes, we are marshaling some types (those are not the proto msg)
Can I use `json.Marshal and json.UnMarshal` instead of `legacyCodec` 
@ashwinphatak  @anilCSE

Reply:
https://github.com/gsk967/ethermint/blob/79f1d14c4790c6edb70094900060e2d16d00a774/x/nameservice/keeper/keeper.go#L144-L144

Reply:
https://github.com/gsk967/ethermint/blob/79f1d14c4790c6edb70094900060e2d16d00a774/x/nameservice/keeper/keeper.go#L245

Reply:
Ideally, we define a protobuf message for that too. Is that difficult to implement?

Reply:
no, it is not difficult ,

Reply:
I will create proto messages wherever we are using legacyCodec

Reply:
@ashwinphatak  removed legacyCodec from nameservice module